addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/handler/OpenDaikinAcUnitHandler.java

```diff
@@ -0,0 +1,267 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles communicating with a Daikin air conditioning unit.
+ *
+ * @author Tim Waterhouse - Initial Contribution
+ *
+ */
+public class OpenDaikinAcUnitHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(OpenDaikinAcUnitHandler.class);
+
+    private long refreshInterval;
+
+    private final Client client = ClientBuilder.newClient();
+    private OpenDaikinWebTargets webTargets;
+    private ScheduledFuture<?> pollFuture;
+
+    public OpenDaikinAcUnitHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            switch (channelUID.getId()) {
+                case OpenDaikinBindingConstants.CHANNEL_AC_POWER:
+                    if (command instanceof OnOffType) {
+                        changePower(((OnOffType) command).equals(OnOffType.ON));
+                    } else {
+                        logger.warn("Received command of wrong type for thing '{}' on channel {}",
+                                thing.getUID().getAsString(), channelUID.getId());
+                    }
+                    break;
+                case OpenDaikinBindingConstants.CHANNEL_AC_TEMP:
+                    if (command instanceof DecimalType) {
+                        changeSetPoint(((DecimalType) command).doubleValue());
+                    } else {
+                        logger.warn("Received command of wrong type for thing '{}' on channel {}",
+                                thing.getUID().getAsString(), channelUID.getId());
+                    }
+                    break;
+                case OpenDaikinBindingConstants.CHANNEL_AC_FAN_SPEED:
+                    if (command instanceof StringType) {
+                        changeFanSpeed(ControlInfo.FanSpeed.valueOf(((StringType) command).toString()));
+                    } else {
+                        logger.warn("Received command of wrong type for thing '{}' on channel {}",
+                                thing.getUID().getAsString(), channelUID.getId());
+                    }
+                    break;
+                case OpenDaikinBindingConstants.CHANNEL_AC_FAN_DIR:
+                    if (command instanceof StringType) {
+                        changeFanDir(ControlInfo.FanMovement.valueOf(((StringType) command).toString()));
+                    } else {
+                        logger.warn("Received command of wrong type for thing '{}' on channel {}",
+                                thing.getUID().getAsString(), channelUID.getId());
+                    }
+                    break;
+                case OpenDaikinBindingConstants.CHANNEL_AC_MODE:
+                    if (command instanceof StringType) {
+                        changeMode(ControlInfo.Mode.valueOf(((StringType) command).toString()));
```

Comment:
If you would extract the switch to another method you could return from the method after succes and you would only have to repeat the `else` once. It will also give you shorter methods.